### PR TITLE
Implement `array.array.__contains__`

### PR DESCRIPTION
**Pull request by moreal**
CPython implements `array.__contains__` but RustPython doesn't implement it yet. This pull request implements it.

## CPython

```python
Python 3.10.9 (main, Dec 15 2022, 17:11:09) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import array
>>> array.array('B').__contains__
<method-wrapper '__contains__' of array.array object at 0x1051ce700>
```

## RustPython (before)

```python
Welcome to the magnificent Rust Python 0.2.0 interpreter 😱 🖖
>>>>> import array
>>>>> array.array('B').__contains__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'array' object has no attribute '__contains__'
```

## RustPython (after)

```python
Welcome to the magnificent Rust Python 0.2.0 interpreter 😱 🖖
>>>>> import array
>>>>> array.array('B').__contains__
<bound method array.__contains__ of array('B')>
```

### Related links

 - https://github.com/python/cpython/blob/5f08fe4a2c055880c23c6f9b57ff03005d193bfc/Modules/arraymodule.c#L1176-L1190

### CODE REVIEW

**Comment by moreal** [comment]
I'm not sure why [running tests in macOS is failing](https://github.com/RustPython/RustPython/actions/runs/3978965699/jobs/6821084358) but others succeed. So I open this pull request as ready for review.

**Comment by youknowone** [comment]
The faling tests is looking unrelated

**Review by youknowone — CHANGES_REQUESTED** [review]
Thank you! I left a suggestion for double expect.

File: stdlib/src/array.rs [hunk 1]
```diff
@@ -1160,6 +1160,24 @@ mod array {
...
             ))
         }
+
+        #[pymethod(magic)]
+        fn contains(&self, value: PyObjectRef, vm: &VirtualMachine) -> bool {
+            let array = self.array.read();
+            for i in 0..array.len() {
+                let element = array
+                    .get(i, vm)
+                    .expect("Expected to be checked by array.len() and read lock.")
```
[the remark this answers is not in the document]
Comment (DimitrisJim) [reply stdlib/src/array.rs:1170]: `ArrayContentType` also has an `iter` which can be used here instead, i.e `for element in array.iter().map(|e| e.expect(...))`

[the remark this answers is not in the document]
Comment (moreal) [reply stdlib/src/array.rs]: I applied this suggestion as `for element in array.iter().map(|e| e.expect(...))` way.

**Comment by DimitrisJim** [comment]
lgtm. do you think you could test this a bit in `extra_tests/stdlib_array.py`? Its weird that no test functions in `test_array` seem to check it.

**Comment by moreal** [comment]
> lgtm. do you think you could test this a bit in `extra_tests/stdlib_array.py`? A couple of `in` calls should do the trick, its weird that no test functions in `test_array` seem to check it.

There is `test_array.NumberTest.test_iterationcontains` test to test `[element] in [array]`. And about the `in` expression how works:

> https://docs.python.org/3/reference/expressions.html#not-in
> For user-defined classes which define the __contains__() method, x in y returns True if y.__contains__(x) returns a true value, and False otherwise.
> 
> For user-defined classes which do not define __contains__() but do define __iter__(), x in y is True if some value z, for which the expression x is z or x == z is true, is produced while iterating over y. If an exception is raised dur[in](https://docs.python.org/3/reference/expressions.html#in)g the iteration, it is as if in raised that exception.

So.. I think the `test_iterationcontains` test is enough.

**Review by DimitrisJim — APPROVED** [review]
> There is test_array.NumberTest.test_iterationcontains test to test [element] in [array]. And about the in expression how works

Ah, I missed that! Either way, thanks for adding some manually to `stdlib_array`

**Pull request merged**